File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/MyPackage/MyComposite.cpp
#include "MyComposite.h"
#include "MyComposite.h"
#include <memory>
#include "RandomProvider.h"
#include "IMyInterface.h"
#include "NumberComponent.h"

namespace MyPackage
{
  MyComposite::MyComposite()
  {
    this->numberComponent = std::make_shared<NumberComponent>();
    this->randomProvider = std::make_shared<RandomProvider>();
    this->randomProvider->dependency_GetNumber = this->numberComponent;
  }
  int MyComposite::GenerateInt(int offset)
  {
    return this->randomProvider->GenerateInt(offset);
  }
  int MyComposite::MyFactor()
  {
    return this->numberComponent->MyFactor();
  }
}

File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/MyPackage/MyComposite.cs
/// <filename>
///     MyComposite.cs
/// </filename>
using MyPackage;

namespace MyPackage
{
    public class MyComposite : MyPackage.IMyInterface
    {

        public MyComposite()
        {
            this.numberComponent = new MyPackage.NumberComponent();
            this.randomProvider = new MyPackage.RandomProvider();
            this.randomProvider.dependency_GetNumber = this.numberComponent;
        }

        public virtual int GenerateInt(int offset)
        {
            return this.randomProvider.GenerateInt(offset);
        }

        public virtual int MyFactor()
        {
            return this.numberComponent.MyFactor();
        }

        public MyPackage.NumberComponent numberComponent;

        public MyPackage.RandomProvider randomProvider;
    }
}
